python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -521,6 +521,11 @@ def op(self) -> Node:
     def op_def(self) -> OpDefinition:
         return self._op_def
 
+    @property
+    def has_assets_def(self) -> bool:
```

Comment:
are both this one and the `has_assets_def` in `context/compute.py` needed? doesn't seem like a bad thing to have both around, just want to understand the code path a bit better